google_guest_agent/telemetry/telemetry.go

```diff
@@ -81,6 +89,64 @@ func Record(ctx context.Context, client metadata.MDSClientInterface, d Data) err
 		"X-Google-Guest-Agent": formatGuestAgent(d),
 		"X-Google-Guest-OS":    formatGuestOS(d),
 	}
-	_, err := client.GetKey(ctx, "project", headers)
+	_, err := client.GetKey(ctx, "", headers)
 	return err
 }
+
+// Job implements job scheduler interface for recording telemetry.
+type Job struct {
+	client       metadata.MDSClientInterface
+	programName  string
+	agentVersion string
+}
+
+// New initializes a new TelemetryJob.
+func New(client metadata.MDSClientInterface, programName, agentVersion string) *Job {
+	return &Job{
+		client:       client,
+		programName:  programName,
+		agentVersion: agentVersion,
+	}
+}
+
+// ID returns the ID for this job.
+func (j *Job) ID() string {
+	return telemetryJobID
+}
+
+// Run records telemetry data.
+func (j *Job) Run(ctx context.Context) (bool, error) {
+	osInfo := osinfo.Get()
+	d := Data{
+		AgentName:     j.programName,
+		AgentVersion:  j.agentVersion,
+		AgentArch:     runtime.GOARCH,
+		OS:            runtime.GOOS,
+		LongName:      osInfo.PrettyName,
+		ShortName:     osInfo.OS,
+		Version:       osInfo.VersionID,
+		KernelRelease: osInfo.KernelRelease,
+		KernelVersion: osInfo.KernelVersion,
+	}
+	if err := Record(ctx, j.client, d); err != nil {
+		// Log this here in Debug mode as telemetry is best effort.
+		logger.Debugf("Error recording telemetry: %v", err)
+	}
+
+	return j.ShouldEnable(ctx), nil
+}
+
+// Interval returns the interval at which job is executed.
+func (j *Job) Interval() (time.Duration, bool) {
+	return telemetryInterval, true
+}
+
+// ShouldEnable returns true as long as DisableTelemetry is not set in metadata.
+func (j *Job) ShouldEnable(ctx context.Context) bool {
+	md, err := j.client.Get(ctx)
+	if err != nil {
+		return false
+	}
+
+	return !md.Instance.Attributes.DisableTelemetry && !md.Project.Attributes.DisableTelemetry
```

Comment:
I think we can flip the logic to default=TRUE now since we have verified the  behaviors on prod and the launch is approved

Reply:
Lets get that into its own PR, as soon as this is Ill send another to change that.